File: clients/java/src/main/java/com/vibium/Browser.java
package com.vibium;

import com.vibium.bidi.BiDiClient;
import com.vibium.process.ClickerProcess;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public final class Browser {
    private static final Logger log = LoggerFactory.getLogger(Browser.class);

    private Browser() {}

    public static Vibe launch() {
        return launch(new LaunchOptions());
    }

    public static Vibe launch(LaunchOptions options) {
        LaunchOptions resolved = options == null ? new LaunchOptions() : options;
        log.info("Launching Vibium (headless={}, port={})", resolved.isHeadless(), resolved.getPort() == null ? "auto" : resolved.getPort());

        ClickerProcess process = ClickerProcess.start(resolved);
        BiDiClient client = null;
        try {
            String wsUrl = String.format("ws://localhost:%d", process.getPort());
            log.debug("Clicker listening at {}", wsUrl);
            client = BiDiClient.connect(wsUrl, resolved.getTimeoutMs());
            return new Vibe(client, process);
        } catch (Exception e) {
            if (client != null) {
                try {
                    client.close();
                } catch (Exception ignored) {
                }
            }
            try {
                process.stop();
            } catch (Exception ignored) {
            }
            throw new VibiumException("Failed to launch Vibium", e);
        }
    }
}

File: clients/javascript/src/browser.ts
import { ClickerProcess } from './clicker';
import { BiDiClient } from './bidi';
import { Vibe } from './vibe';
import { debug, info } from './utils/debug';

export interface LaunchOptions {
  headless?: boolean;
  port?: number;
  executablePath?: string;
  proxy?: string;
}

export const browser = {
  async launch(options: LaunchOptions = {}): Promise<Vibe> {
    const { headless = false, port, executablePath, proxy } = options;
    debug('launching browser', { headless, port, executablePath, proxy });

    // Start the clicker process
    const process = await ClickerProcess.start({
      headless,
      port,
      executablePath,
      proxy,
    });
    debug('clicker started', { port: process.port });

    // Connect to the proxy
    const client = await BiDiClient.connect(`ws://localhost:${process.port}`);
    info('browser launched', { port: process.port });

    return new Vibe(client, process);
  },
};
